Xamarin.PropertyEditing/CategoryComparer.cs

```diff
@@ -0,0 +1,20 @@
+using System;
+using System.Collections.Generic;
+
+namespace Xamarin.PropertyEditing
+{
+	internal class CategoryComparer
+		: IComparer<string>
+	{
+		public static readonly CategoryComparer Instance = new CategoryComparer();
+
+		public int Compare (string x, string y)
+		{
+			int result = Comparer<string>.Default.Compare (x, y);
+			if (result != 0 && (String.IsNullOrEmpty (x) || String.IsNullOrEmpty (y)))
```

Comment:
It would be good to doc-comment how this differs from the default string comparison, which is if I understand correctly that null goes last.